### ISSUE TITLE
It takes much longer time to boot a vm when using distributed minio

### ISSUE THREAD

**Issue by gitlawr** [harvester/harvester#253]
When minio runs in distributed mode, the time for CDI to download an ubuntu image in hundreds of megabytes exceeds over 3 minutes. As the comparison, it takes ~30s in standalone mode.

**Comment by gitlawr** [comment]
Most of the time is used in the convert phase of CDI import. It calls `qemu-img convert` under the hood.
Downloading the image from Minio directly is fast though.

The following tests are based on a 345M [ubuntu image](https://cloud-images.ubuntu.com/releases/bionic/release-20201117/ubuntu-18.04-server-cloudimg-amd64.img).

Distributed mode Minio:
```
# convert from the URL directly (the way CDI uses)
$ time qemu-img convert -t none -p -O raw "json: {\"file.driver\":\"http\",\"file.url\":\"http://minio.harvester-system:9000/vm-images/image-48jc9\"}" /disk1.img
real	4m34.108s
user	0m9.072s
sys	0m6.608s

# download the image then do convert locally
$ time curl -O http://minio.harvester-system:9000/vm-images/image-48jc9
real	0m2.680s
user	0m0.044s
sys	0m0.784s
$ time qemu-img convert -t none -p -O raw image-48jc9 /disk2.img
real	0m24.889s
user	0m7.168s
sys	0m2.352s
```

Standalone mode Minio:
```
# convert from the URL directly (the way CDI uses)
$ time qemu-img convert -t none -p -O raw "json: {\"file.driver\":\"http\",\"file.url\":\"http://minio.harvester-system:9000/vm-images/image-55c4j\"}" /disk1.img
real	0m22.430s
user	0m7.196s
sys	0m2.700s

# download the image then do convert locally
$ time curl -O http://minio.harvester-system:9000/vm-images/image-55c4j
real	0m2.896s
user	0m0.080s
sys	0m0.664s
$ time qemu-img convert -t none -p -O raw image-55c4j /disk2.img
real	0m24.666s
user	0m6.792s
sys	0m2.736s
```

**Comment by gitlawr** [comment]
I did some performance tests using both modes of Minio. In summary:
1. It cost similar time getting a single image file: 4.650s (standalone) vs 4.574s (distributed)
2. Operations of a number of small objects: distributed minio has lower throughput(4.81 MiB/s, 492.80 obj/s) than standalone minio(26.29 MiB/s, 2692.19 obj/s).

The following performance tests use the [Warp](https://github.com/minio/warp) tool.

```
warp get --host=STANDALONE --access-key=x --secret-key=x --autoterm --obj.size=10KiB
Operation: PUT
* Average: 8.74 MiB/s, 895.22 obj/s

Throughput, split into 2 x 1s:
 * Fastest: 9.2MiB/s, 942.84 obj/s
 * 50% Median: 9.2MiB/s, 942.84 obj/s
 * Slowest: 9.0MiB/s, 925.65 obj/s

-------------------

Operation: GET
* Average: 26.29 MiB/s, 2692.19 obj/s

Throughput, split into 299 x 1s:
 * Fastest: 38.1MiB/s, 3900.34 obj/s
 * 50% Median: 31.2MiB/s, 3191.19 obj/s
 * Slowest: 807.9KiB/s, 80.80 obj/s
```

```
$ warp get --host=DISTRIBUTED --access-key=x --secret-key=x --autoterm --obj.size=10KiB
Operation: PUT
* Average: 0.54 MiB/s, 55.37 obj/s

Throughput, split into 44 x 1s:
 * Fastest: 863.4KiB/s, 86.34 obj/s
 * 50% Median: 593.7KiB/s, 59.37 obj/s
 * Slowest: 112.2KiB/s, 11.22 obj/s

-------------------

Operation: GET
* Average: 4.81 MiB/s, 492.80 obj/s

Throughput, split into 145 x 1s:
 * Fastest: 6.5MiB/s, 665.20 obj/s
 * 50% Median: 5.2MiB/s, 535.54 obj/s
 * Slowest: 1404.2KiB/s, 140.43 obj/s
```

Regarding the `qemu-img convert` implementation, if the target is an HTTP URL, it performs thousands of [HTTP range](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range) requests resulting in poor performance using distributed Minio.

**Comment by gitlawr** [comment]
So it would be proper to download the image then convert it locally. 
The [CDI code](https://github.com/kubevirt/containerized-data-importer/blob/09b61a924c66869cfd9693ed5600b6a2caa3a4c4/pkg/importer/http-datasource.go#L108-L129) shows that it will download then do convert in certain cases when using the HTTP source.

**Comment by guangbochen** [comment]
validation steps:
1. creating the VM using a ubuntu image
2. get pvc and found 2 volumes has been created for the vm
```
pvc-63cf9f4f-3769-4e9d-ba0f-bf26f7c8eb10   10Gi       RWO            Delete           Bound    default/ub1-disk-0-mwwzj-scratch   longhorn                112s
pvc-35489c1a-6238-417f-b934-3a16e160a6d4   10Gi       RWO            Delete           Bound    default/ub1-disk-0-mwwzj           longhorn                106s
```
3. check pod logs and found the image import and download are complete in a couple of seconds as expected.
```
I1216 17:28:11.231233       1 importer.go:52] Starting importer
I1216 17:28:11.231328       1 importer.go:117] begin import process
I1216 17:28:11.284541       1 data-processor.go:296] Calculating available size
I1216 17:28:11.284743       1 data-processor.go:308] Checking out file system volume size.
I1216 17:28:11.284861       1 data-processor.go:316] Request image size not empty.
I1216 17:28:11.284959       1 data-processor.go:321] Target size 10447220736.
I1216 17:28:11.285131       1 util.go:37] deleting file: /scratch/lost+found
I1216 17:28:11.292869       1 util.go:37] deleting file: /data/lost+found
I1216 17:28:11.320974       1 data-processor.go:218] New phase: TransferScratch
I1216 17:28:11.321696       1 util.go:161] Writing data...
I1216 17:28:12.843200       1 data-processor.go:218] New phase: Convert
I1216 17:28:12.843229       1 data-processor.go:224] Validating image
I1216 17:28:29.146552       1 data-processor.go:218] New phase: Resize
W1216 17:28:29.153541       1 data-processor.go:283] Available space less than requested size, resizing image to available space 10447220736.
I1216 17:28:29.153562       1 data-processor.go:289] Expanding image size to: 10447220736
I1216 17:28:29.160239       1 data-processor.go:218] New phase: Complete
I1216 17:28:29.160296       1 util.go:37] deleting file: /scratch/tmpimage
I1216 17:28:29.183070       1 importer.go:190] Import complete
```
